core/domain/collection_services.py

```diff
@@ -0,0 +1,872 @@
+
+    If run_conversion is True, then the collection's schema version will be
+    checked against the current schema version. If they do not match, the
+    collection will be automatically updated to the latest schema version.
+
+    IMPORTANT NOTE TO DEVELOPERS: In general, run_conversion should never be
+    False. This option is only used for testing that the schema version
+    migration works correctly, and it should never be changed otherwise.
+    """
+
+    # Ensure the original collection model does not get altered.
+    versioned_collection = {
+        'schema_version': collection_model.schema_version,
+        'nodes': copy.deepcopy(collection_model.nodes)
+    }
+
+    # Migrate the collection if it is not using the latest schema version.
+    if (run_conversion and collection_model.schema_version !=
+            feconf.CURRENT_COLLECTION_SCHEMA_VERSION):
+        _migrate_collection_to_latest_schema(versioned_collection)
+
+    return collection_domain.Collection(
+        collection_model.id, collection_model.title,
+        collection_model.category, collection_model.objective,
+        versioned_collection['schema_version'], [
+            collection_domain.CollectionNode.from_dict(collection_node_dict)
+            for collection_node_dict in versioned_collection['nodes']
+        ],
+        collection_model.version, collection_model.created_on,
+        collection_model.last_updated)
+
+
+def get_collection_summary_from_model(collection_summary_model):
+    return collection_domain.CollectionSummary(
+        collection_summary_model.id, collection_summary_model.title,
+        collection_summary_model.category, collection_summary_model.objective,
+        collection_summary_model.status,
+        collection_summary_model.community_owned,
+        collection_summary_model.owner_ids,
+        collection_summary_model.editor_ids,
+        collection_summary_model.viewer_ids,
+        collection_summary_model.version,
+        collection_summary_model.collection_model_created_on,
+        collection_summary_model.collection_model_last_updated
+    )
+
+
+def get_collection_by_id(collection_id, strict=True, version=None):
+    """Returns a domain object representing a collection."""
+    collection_memcache_key = _get_collection_memcache_key(
+        collection_id, version=version)
+    memcached_collection = memcache_services.get_multi(
+        [collection_memcache_key]).get(collection_memcache_key)
+
+    if memcached_collection is not None:
+        return memcached_collection
+    else:
+        collection_model = collection_models.CollectionModel.get(
+            collection_id, strict=strict, version=version)
+        if collection_model:
+            collection = get_collection_from_model(collection_model)
+            memcache_services.set_multi({collection_memcache_key: collection})
+            return collection
+        else:
+            return None
+
+
+def get_collection_summary_by_id(collection_id):

```

Comment:
> [@seanlip] Recording offline discussion: let's keep the summaries separate for now, but add a TODO to merge things (and perhaps generate them with a batch job) if there is no divergence by Jan 2016 (say). Also, file an issue for this. We may also want to recharacterize summaries as SearchSnippets or similar, since that's really what they are.

> > [@BenHenning] I may not remember correctly, but I do not think the offline discussion was particularly picky about keeping the summaries separate. Since the changes in rights_manager introduced the concept of an activity (and related activity types for collections and explorations), I'm inclined to unify the two summaries into an ActivitySummary and differentiate between the two summaries based on their activity type (though this might not even be needed since the summaries are differentiated based on their search indexes, anyway).

SGTM.